src/iuse.cpp

```diff
@@ -2194,122 +2194,98 @@ void remove_battery_mods( item &modded, player &p )
 int iuse::extra_battery(player *p, item *, bool, const tripoint& )
 {
     int inventory_index = g->inv_for_tools_powered_by( ammotype( "battery" ), _( "Modify what?" ) );
-    item *modded = &( p->i_at( inventory_index ) );
+    item &modded = p->i_at( inventory_index );
 
-    if (modded == NULL || modded->is_null()) {
+    if( modded.is_null() ) {
         p->add_msg_if_player(m_info, _("You do not have that item!"));
         return 0;
     }
 
-    if (modded->has_flag("DOUBLE_AMMO")) {
+    if (modded.has_flag("DOUBLE_AMMO")) {
         p->add_msg_if_player(m_info, _("That item has already had its battery capacity doubled."));
         return 0;
     }
 
-    remove_battery_mods( *modded, *p );
+    remove_battery_mods( modded, *p );
 
-    p->add_msg_if_player( _( "You double the battery capacity of your %s!" ), modded->tname().c_str() );
-    modded->item_tags.insert("DOUBLE_AMMO");
+    p->add_msg_if_player( _( "You double the battery capacity of your %s!" ), modded.tname().c_str() );
+    modded.item_tags.insert("DOUBLE_AMMO");
     return 1;
 }
 
 int iuse::double_reactor(player *p, item *, bool, const tripoint& )
 {
     int inventory_index = g->inv_for_tools_powered_by( ammotype( "plutonium" ), _( "Modify what?" ) );
-    item *modded = &( p->i_at( inventory_index ) );
+    item &modded = p->i_at( inventory_index );
 
-    if (modded == NULL || modded->is_null()) {
+    if( modded.is_null() ) {
         p->add_msg_if_player(m_info, _("You do not have that item!"));
         return 0;
     }
 
-    p->add_msg_if_player( _( "You double the plutonium capacity of your %s!" ), modded->tname().c_str() );
-    modded->item_tags.insert("DOUBLE_AMMO");
-    modded->item_tags.insert("DOUBLE_REACTOR");   //This flag lets the remove_ functions know that this is a plutonium tool without taking extra steps.
-    return 1;
-}
-
-int iuse::rechargeable_battery(player *p, item *it, bool, const tripoint& )
```

Comment:
This is more than just refactoring to use the relevant `item` constructor and `ammo_*` accessors?